### ISSUE TITLE
Indicate dropdown selection with aria-selected attribute

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/exhibits#2499]
### Current behavior

`aria-current` is used to indicate a dropdown selection. Use `aria-selected` instead.

### Steps to reproduce

Example page: [Spotlight Accessibility for SODA - San Francisco Buildings](https://exhibits.stanford.edu/a11y-for-soda/browse/san-francisco-buildings).

Click on one of the dropdown menu options ("Sort by relevance," "12 per page").

### Possible solution

WCAG Level A

See SODA's recommendation below.

### Reported by SODA

> Violation: Avoid inappropriate use of ARIA roles, states, and properties
> Severity: Minor
> 
> [Issue]
> The dropdown for the sorting uses aria-current to indicate which is selected. The appropriate attribute here is aria selected. 
> 
> Note that aria-current is used for the the page selector at the bottom and is appropriate there. 
> 
> [User Impact]
> When inappropriate ARIA roles, states, and/or properties are used it may prevent users of assistive technology from correctly interpreting and interacting with the content.
> 
> [Code Reference]
> ```
> <div class="dropdown-menu show" role="menu" x-placement="bottom-start" style="...">
>   <a class="dropdown-item " role="menuitem" href="...">relevance</a>
>   <a class="dropdown-item " role="menuitem" href="...">year (new to old)</a>
>   <a class="dropdown-item active" role="menuitem" aria-current="page" href="...">year (old to new)</a>
>   <a class="dropdown-item " role="menuitem" href="...">author</a>
>   <a class="dropdown-item " role="menuitem" href="...">title</a>
>   <a class="dropdown-item " role="menuitem" href="...">publisher</a>
> </div>
> 
> ```
> [Recommendation]
> Change aria-current to aria-selected="true".

**Comment by jcoyne** [comment]
Is this because we are combining links and role="menuitem"?  I believe aria-current is appropriate for links, no?

**Comment by taylor-steve** [comment]
I agree with @jcoyne. Role there is `menuitem`. Spec does say:

> Authors SHOULD NOT use the [aria-current](https://w3c.github.io/aria/#aria-current) attribute as a substitute for [aria-selected](https://w3c.github.io/aria/#aria-selected) in widgets where [aria-selected](https://w3c.github.io/aria/#aria-selected) has the same meaning.


However, looking at [aria-selected](https://w3c.github.io/aria/#aria-selected), menu/menuitem doesn't seem to fall under that.

The [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected) seem to confirm:
>The aria-selected attribute indicates the current "selected" state for 
> [gridcell](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/gridcell_role), [option](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/option_role), [row](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/row_role) and [tab](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role) roles.
> 
> This attribute is used to indicate which elements within single-selection and multiple-selection composite widgets are selected. If more than one element is selectable at a time, include aria-multiselectable="true" on the grid, listbox, tablist, or other owning role, while including aria-selected only on the selectable cells, options, and tabs.
> 
> For other roles, the currently selected state is set with [aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current), or possibly [aria-checked](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) or [aria-pressed](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed), depending on the role.

The [menuitem role](https://w3c.github.io/aria/#menuitem) spec only lists `aria-current`, it does not state support for `aria-selected`.

@alundgard what's your take? What do you recommend?

**Comment by alundgard** [comment]
@taylor-steve Looking into the ARIA documentation more, I agree with you. It looks like `aria-current` is appropriate for the menuitem role — although there are other, similar roles (such as treeitem) where `aria-selected` would be appropriate. My guess is that SODA might've had these other roles in mind when making this recommendation.
